SRS/SRS_OC_BODBAW/srs_oc_bodbaw_netcdf_creation.py

```diff
@@ -219,19 +219,22 @@ def data_frame_ac9_hs6(self):
 
     def data_frame_absorption(self):
         """ return data as a pandas data frame """
+        n_var_rows_defined = self.idx_end_var_rows_def - self.idx_start_var_rows_def - 1
         data = []
-        for i in range(self.idx_start_data + 6, self.idx_end_data):
+        for i in range(self.idx_start_data + n_var_rows_defined + 1, self.idx_end_data):
             data.append(self.sheet.row_at(i)[2:self.max_data_col + 2])
         data_frame = pd.DataFrame(data)
 
         dates                 = self.sheet.row_at(self.idx_start_data + 1)[2:self.max_data_col + 2]
         data_frame.Dates      = pd.to_datetime(dates)
-        data_frame.Wavelength = self.sheet.column_at(1)[self.idx_start_data + 6:]
+        data_frame.Wavelength = self.sheet.column_at(1)[self.idx_start_data + n_var_rows_defined + 1:]
 
-        data_frame.Station_Code = self.sheet.row_at(self.idx_start_data + 2)[2:self.max_data_col + 2]
-        data_frame.Latitude     = self.sheet.row_at(self.idx_start_data + 3)[2:self.max_data_col + 2]
-        data_frame.Longitude    = self.sheet.row_at(self.idx_start_data + 4)[2:self.max_data_col + 2]
-        data_frame.Depth        = self.sheet.row_at(self.idx_start_data + 5)[2:self.max_data_col + 2]
+        data_frame.Station_Code = self.sheet.row_at(self.idx_start_data + self.sheet.column_at(0)[self.idx_start_data:].index('Station_Code'))[2:self.max_data_col + 2]
+        data_frame.Latitude     = self.sheet.row_at(self.idx_start_data + self.sheet.column_at(0)[self.idx_start_data:].index('Latitude'))[2:self.max_data_col + 2]
+        data_frame.Longitude    = self.sheet.row_at(self.idx_start_data + self.sheet.column_at(0)[self.idx_start_data:].index('Longitude'))[2:self.max_data_col + 2]
+        data_frame.Depth        = self.sheet.row_at(self.idx_start_data + self.sheet.column_at(0)[self.idx_start_data:].index('Depth'))[2:self.max_data_col + 2]
+        if 'Sample_Number' in self.sheet.column_at(0)[self.idx_start_data:]:
+            data_frame.Sample_Number = self.sheet.row_at(self.idx_start_data + self.sheet.column_at(0)[self.idx_start_data:].index('Sample_Number'))[2:self.max_data_col + 2]
```

Comment:
You could make these lines slightly shorter and easier to read by assigning 
```python
self.sheet.column_at(0)[self.idx_start_data:]
```
to a local variable with a shorter name?